### PR TITLE
fix convert issue

### PR DESCRIPTION
**Pull request by UncleTom**
if  attribuets/parameters is {}, it will not do convert action, but it should be convert to [].

### CODE REVIEW

File: reportportal_client/service.py [hunk 1]
```diff
@@ -207,7 +207,7 @@ def start_launch(self,
                      mode=None,
                      **kwargs):
         """Start a new launch with the given parameters."""
-        if attributes is not None:
+        if isinstance(attributes, dict):
```
Comment (iivanou) [inline reportportal_client/service.py]: I'd rather make a change in the convert function itself.

```
def _dict_to_payload(dictionary):
    """Convert dict to list of dicts.

    :param dictionary: initial dict
    :return list: list of dicts
    """
    if not isinstance(dictionary, dict):
        return None
    system = dictionary.pop("system", False)
    return [
        {"key": key, "value": _convert_string(value), "system": system}
        for key, value in sorted(dictionary.items())
    ]
```

Do not forget about unit tests.

Comment (UncleTom) [reply reportportal_client/service.py]: if attributes is {}, it does't go the _dict_to_payload function, and directly post to the API with {} value.
do we need re-back the condition to : if attributes is not None  ?

File: reportportal_client/service.py [hunk 2]
```diff
@@ -262,9 +264,9 @@ def start_test_item(self,
                 ...
             }
         """
-        if attributes:
+        if attributes is not None:
```
Comment (iivanou) [inline reportportal_client/service.py]:
```suggestion
        if attributes:
```

File: reportportal_client/service.py [hunk 3]
```diff
@@ -262,9 +264,9 @@ def start_test_item(self,
...
             attributes = _dict_to_payload(attributes)
-        if parameters:
+        if parameters is not None:
```
Comment (iivanou) [inline reportportal_client/service.py]:
```suggestion
        if parameters:
```

File: reportportal_client/service.py [hunk 4]
```diff
@@ -310,7 +312,7 @@ def finish_test_item(self,
                 and not self.is_skipped_an_issue:
             issue = {"issue_type": "NOT_ISSUE"}
 
-        if attributes:
+        if attributes is not None:
```
Comment (iivanou) [inline reportportal_client/service.py]:
```suggestion
        if attributes:
```

File: tests/test_service.py [hunk 5]
```diff
@@ -36,6 +36,11 @@ def test_dict_to_payload_with_system_key(self, system):
                          {'key': 'b', 'value': '2', 'system': system}]
         assert _dict_to_payload(initial_dict) == expected_list
 
+    @pytest.mark.parametrize('input, expected', [(None, None), ({}, None)])
+    def test_dict_to_payload_with_empty(self, input, expected):
+        """Test convert empty/None dict to list of dicts."""
+        assert _dict_to_payload(input) == expected
```
Comment (iivanou) [inline tests/test_service.py]:
```
    def test_dict_to_payload_with_not_dict(self):
        """Test _dict_to_payload passed with NOT a dictionary."""
        assert _dict_to_payload('hello') == None
```

File: reportportal_client/service.py [hunk 6]
```diff
@@ -49,6 +49,8 @@ def _dict_to_payload(dictionary):
     :param dictionary: initial dict
     :return list: list of dicts
     """
+    if not isinstance(dictionary, dict) or len(dictionary) == 0:
```
Comment (iivanou) [inline reportportal_client/service.py]:
```suggestion
    if not isinstance(dictionary, dict):
```

**Comment by UncleTom** [comment]
please see below code:

```python
attributes = {}
if attributes:
    print('it will not go here')
    # attributes = _dict_to_payload(attributes)
else:
    print('it will go here')
```
if attributes = {}, the final value is {}, but not [] or None.  API should response error.

**Pull request closed without merging**